File: backend/app/schemas/work_order.py
import uuid
from datetime import datetime

from app.models.work_order import WorkOrderPriority, WorkOrderStatus
from app.schemas.base import CustomBaseModel, PaginationMeta


class WorkOrderCreate(CustomBaseModel):
    title: str
    description: str | None = None
    priority: WorkOrderPriority = WorkOrderPriority.MEDIUM
    client_id: uuid.UUID | None = None
    technician_id: uuid.UUID | None = None
    scheduled_at: datetime | None = None


class WorkOrderUpdate(CustomBaseModel):
    title: str | None = None
    description: str | None = None
    priority: WorkOrderPriority | None = None
    client_id: uuid.UUID | None = None
    technician_id: uuid.UUID | None = None
    scheduled_at: datetime | None = None


class WorkOrderRead(CustomBaseModel):
    id: uuid.UUID
    title: str
    description: str | None
    status: WorkOrderStatus
    priority: WorkOrderPriority
    client_id: uuid.UUID | None
    technician_id: uuid.UUID | None
    scheduled_at: datetime | None
    completed_at: datetime | None
    created_at: datetime
    updated_at: datetime


class StatusTransitionRequest(CustomBaseModel):
    to_status: WorkOrderStatus
    notes: str | None = None


class WorkOrderStatusHistoryRead(CustomBaseModel):
    id: uuid.UUID
    work_order_id: uuid.UUID
    from_status: WorkOrderStatus | None
    to_status: WorkOrderStatus
    notes: str | None
    created_at: datetime


class WorkOrdersResponse(CustomBaseModel):
    work_orders: list[WorkOrderRead]
    meta: PaginationMeta

File: frontend/src/types/work-order.ts
export type WorkOrderStatus =
  | 'pending'
  | 'assigned'
  | 'in_progress'
  | 'completed'
  | 'cancelled'

export type WorkOrderPriority = 'low' | 'medium' | 'high' | 'urgent'

export interface WorkOrder {
  id: string
  title: string
  description: string | null
  status: WorkOrderStatus
  priority: WorkOrderPriority
  client_id: string | null
  technician_id: string | null
  scheduled_at: string | null
  completed_at: string | null
  created_at: string
  updated_at: string
}

export interface WorkOrderCreate {
  title: string
  description?: string | null
  priority?: WorkOrderPriority
  client_id?: string | null
  technician_id?: string | null
  scheduled_at?: string | null
}

export interface WorkOrderUpdate {
  title?: string | null
  description?: string | null
  priority?: WorkOrderPriority | null
  client_id?: string | null
  technician_id?: string | null
  scheduled_at?: string | null
}

export interface StatusTransitionRequest {
  to_status: WorkOrderStatus
  notes?: string | null
}

export interface WorkOrderStatusHistory {
  id: string
  work_order_id: string
  from_status: WorkOrderStatus | null
  to_status: WorkOrderStatus
  notes: string | null
  created_at: string
}

export interface WorkOrderFilters {
  status?: WorkOrderStatus
  priority?: WorkOrderPriority
  technician_id?: string
  client_id?: string
  search?: string
  sort_by?: string
  sort_dir?: 'asc' | 'desc'
  page?: number
  page_size?: number
}

export interface WorkOrdersResponse {
  work_orders: WorkOrder[]
  meta: import('./client').PaginationMeta
}
